src/puppetlabs/puppetdb/http/catalogs.clj

```diff
@@ -48,10 +48,12 @@
                       (str (:url-prefix globals))))
 
     [node "edges" &]
-    (comp (edges/edges-app version false) (partial http-q/restrict-query-to-node node))
+    (-> (comp (edges/edges-app version false) (partial http-q/restrict-query-to-node node))
+        (wrap-with-parent-check version :node node))
 
     [node "resources" &]
-    (comp (resources/resources-app version false) (partial http-q/restrict-query-to-node node))))
+    (-> (comp (resources/resources-app version false) (partial http-q/restrict-query-to-node node))
+        (wrap-with-parent-check version :node node))))
 
 (defn catalog-app
   [version]
```

Comment:
actually, this is a bit more tricky than I had assumed, because the two checks are being done a bit differently. For a node that exists, but has no catalogs, this is our current behavior:

```
curl -X GET http://localhost:8080/pdb/query/v4/catalogs/desktop.localdomain
{
  "error" : "Could not find catalog for desktop.localdomain"
}
```

``
curl -X GET http://localhost:8080/pdb/query/v4/catalogs/desktop.localdomain/resources
[ ]
```

This seems inconsistent to me, but I'm not sure what the right approach is. It feels like something should be thrown in the first case, but in the second case, is the parent the node, or "node with a catalog"?